feat: Add leader election metrics (#56)
kyma-project/opentelemetry-collector-components: receiver/singletonreceivercreator/documentation.md

[comment]: <> (Code generated by mdatagen. DO NOT EDIT.)

# singleton_receiver_creator

## Internal Telemetry

The following telemetry is emitted by this component.

### receiver_singleton_leader_status

> Gauge of if the reporting system is leader of the relevant lease, 0 indicates backup, 1 indicates leader.

Reviewer: ```suggestion
A gauge of if the reporting system is the leader of the relevant lease, 0 indicates backup, and 1 indicates leader.
```

| Unit | Metric Type | Value Type |
| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

### receiver_singleton_lease_acquired_total

> Total number of successful lease acquisitions.

Reviewer: ```suggestion
The total number of successful lease acquisitions.
```

| Unit | Metric Type | Value Type | Monotonic |
| ---- | ----------- | ---------- | --------- |
| 1 | Sum | Int | true |

### receiver_singleton_lease_lost_total

> Total number of lease losses.

Reviewer: ```suggestion
The total number of lease losses.
```

| Unit | Metric Type | Value Type | Monotonic |
| ---- | ----------- | ---------- | --------- |
| 1 | Sum | Int | true |

### receiver_singleton_lease_slowpath_total

> Total number of slow path exercised in renewing leader leases.

Reviewer: ```suggestion
The total number of slow paths exercised in renewing leader leases.
```
